src/model/test/Component_GTest.cpp

```diff
@@ -463,3 +467,98 @@ TEST_F(ModelFixture, Component_Purge_NonResourceObject) {
+    Model model;
+    ASSERT_EQ(0u, model.numObjects());
+
+    Model model2;
+    SiteWaterMainsTemperature siteWaterMainsTemperature2 = model2.getUniqueModelObject<SiteWaterMainsTemperature>();
+    ASSERT_EQ(1u, model2.numObjects());
+
+    // create component
+    Component siteWaterMainsTemperature2Component = siteWaterMainsTemperature2.createComponent();
+    EXPECT_EQ(1u, siteWaterMainsTemperature2Component.componentData().numComponentObjects());
+    EXPECT_EQ(1u, model2.numObjects());
+
+    // insert component
+    OptionalComponentData ocd = model.insertComponent(siteWaterMainsTemperature2Component);
+    ASSERT_TRUE(ocd);
+    EXPECT_NE(siteWaterMainsTemperature2Component.componentData().handle(), ocd->handle());
+    EXPECT_EQ(1u, ocd->numComponentObjects());
+    ASSERT_TRUE(ocd->primaryComponentObject().optionalCast<SiteWaterMainsTemperature>());
+
+    // check model
+    EXPECT_EQ(1u, model.getModelObjects<SiteWaterMainsTemperature>().size());
+    ASSERT_EQ(1u, model.getModelObjects<ComponentData>().size());
+    EXPECT_EQ(ocd->handle(), model.getModelObjects<ComponentData>()[0].handle());
+    EXPECT_EQ(2u, model.numObjects());
+    EXPECT_NE(siteWaterMainsTemperature2.handle(), model.getUniqueModelObject<SiteWaterMainsTemperature>().handle());  // cloned
+  }
+
+  {
+    // insert component with SiteWaterMainsTemperature object into model with SiteWaterMainsTemperature object
+    Model model;
+    SiteWaterMainsTemperature siteWaterMainsTemperature1 = model.getUniqueModelObject<SiteWaterMainsTemperature>();
+    ASSERT_EQ(1u, model.numObjects());
+
+    Model model2;
+    SiteWaterMainsTemperature siteWaterMainsTemperature2 = model2.getUniqueModelObject<SiteWaterMainsTemperature>();
+    ASSERT_EQ(1u, model2.numObjects());
+
+    // create component
+    Component siteWaterMainsTemperature2Component = siteWaterMainsTemperature2.createComponent();
+    EXPECT_EQ(1u, siteWaterMainsTemperature2Component.componentData().numComponentObjects());
+    EXPECT_EQ(1u, model2.numObjects());
+
+    // insert component
+    OptionalComponentData ocd = model.insertComponent(siteWaterMainsTemperature2Component);
+    ASSERT_TRUE(ocd);
+    EXPECT_NE(siteWaterMainsTemperature2Component.componentData().handle(), ocd->handle());
+    EXPECT_EQ(1u, ocd->numComponentObjects());
+    ASSERT_TRUE(ocd->primaryComponentObject().optionalCast<SiteWaterMainsTemperature>());
+
+    // check model
+    EXPECT_EQ(1u, model.getModelObjects<SiteWaterMainsTemperature>().size());
+    ASSERT_EQ(1u, model.getModelObjects<ComponentData>().size());
+    EXPECT_EQ(ocd->handle(), model.getModelObjects<ComponentData>()[0].handle());
+    EXPECT_EQ(2u, model.numObjects());
+    EXPECT_NE(siteWaterMainsTemperature1.handle(), model.getUniqueModelObject<SiteWaterMainsTemperature>().handle());  // removed
+    EXPECT_NE(siteWaterMainsTemperature2.handle(), model.getUniqueModelObject<SiteWaterMainsTemperature>().handle());  // cloned
```

Comment:
Please extend the test to to make sure that the resulting object does have the right fields.